### ISSUE TITLE
Circuit Relay Testing - Failures + Needs fixing

### ISSUE THREAD

**Issue by dryajov** [ipfs/interop#11]
While testing circuit-relay interop functionality, I've come across a number of failures that seem to happen randomly. I'll try to document those here as a first step, which will most likely end up as their own issues once properly investigated.

- `  1) circuit,js relay,js <-> js relay <-> go
       should connect:
     Error: Error: MAC Invalid: 162229cc4f2e3245537de8286c8a82c80ae3f2b31b6e2cb5287205664bd0de9e != 90eea8a5f0839cb96e5b661f904285dd4437dabab1752b3559e83858b5b0ce08`
	- affected OSs
		- windows v8.9.1 and v9.2.0
		- linux  v9.2.0 and v8.9.1
		- macos v8.9.1 and 9.2.0
	- affected CIs
		- jenkins
		- Circleci
		- Travis

- `Firefox 58.0.0 (Linux 0.0.0) ERROR
  Disconnected, because no message in 500000 ms.
  25 passing (14m)
  6 pending
Some tests are failing`
	- affected OSs
		- windows v8.9.1 and v9.2.0
		- linux  v9.2.0 and v8.9.1
		- macos v8.9.1 and 9.2.0
	- affected CIs
		- jenkins
		- Circleci
		- Travis

- `  1) circuit go relay go <-> go relay <-> go should connect:
     Error: connect QmPESBd38krVFxtykakWtNzNTn9wAfGDry9nkjSzYcJe1R failure: dial attempt failed: context deadline exceeded`
	- affected OSs
		- macos v8.9.1 and 9.2.0
	- affected CIs
		- jenkins
		- Circleci
		- Travis		 
	- should be addressed with - https://github.com/libp2p/go-libp2p-circuit/issues/27

**Comment by diasdavid** [comment]
Why two issues? The other being https://github.com/ipfs/interop/issues/12

**Comment by diasdavid** [comment]
Btw, the first looks like a problem in SECIO, the second looks like the connection got closed and so it never finished and the third the same thing but where go-ipfs timed out. This signals that Circuit is not fully finished.

**Comment by dryajov** [comment]
> Why two issues? The other being #12

This was to capture the individual failures that I came across, #12 is here to document the general issues with setup/framework/approach/etc... It wasn't a specific issue, but rather document my overall experience.

> Btw, the first looks like a problem in SECIO

Sure, I've seen this my self intermittently without circuit and so have others it seems - https://github.com/ipfs/js-ipfs/issues/945

> he second looks like the connection got closed and so it never finished and the third the same thing but where go-ipfs timed out

The only way to understand what happened here is to isolate what/why/when closed the connection - been working on that since I saw it happen

> This signals that Circuit is not fully finished.

I don't see anything in the above issues that would point directly to circuit, timeouts/connection drops are existing issues that we intermittently run into, the SESSIO issue is another one that we've seen popup randomly and if I remember correctly, you pointed during a conversation that it might also be due to connection drops as well. 

If you have any ideas on how to troubleshoot this on our CI ENVs, I would appreciate you sharing them.
